lib/common/http1client.c

```diff
@@ -789,30 +827,53 @@ static void start_request(struct st_h2o_http1client_t *client, h2o_iovec_t metho
     size_t reqbufcnt = 0;
     if (props->proxy_protocol->base != NULL)
         reqbufs[reqbufcnt++] = *props->proxy_protocol;
-    h2o_iovec_t header = build_request(client, method, url, *props->connection_header, headers, num_headers);
+    h2o_iovec_t header = build_request(client, method, url, props, headers, num_headers);
     reqbufs[reqbufcnt++] = header;
     client->super.bytes_written.header = header.len;
 
     client->_is_chunked = *props->chunked;
     client->_method_is_head = h2o_memis(method.base, method.len, H2O_STRLIT("HEAD"));
 
+    if (props->expect_100_continue) {
+        if (client->proceed_req == NULL && body.len == 0) {
+            /* no body */
+            on_whole_request_sent(client->sock, h2o_httpclient_error_internal);
+            return;
+        }
```

Comment:
ditto

Reply:
I think we still have a problem here; to me it seems that this if block rejects sending non-streamed HTTP requests with zero-byte of body.

Reply:
Thank you so much! I think the right thing to do here is just do nothing if the req has no body, instead of erroring out. Addressed in https://github.com/h2o/h2o/pull/3316/commits/64aa6217b8cc1d2d64a5f172aeae1b481754788a